Fix most tests

Tomin1 opened the pull request:
Fixes all but two failing cases, which I added comments about to tests.xml but I'm not sure if we should comment them out or try harder to fix them. Anyway this already improves a lot.

Patch test-compress to avoid segfault. Too large stack, thus moving a large array out of stack.

Add upstream patches to test-execute, do some fixes for busybox compatibility and skip some tests that don't have a chance to work.

Adjust tests.xml. I changed some tests to expect skipped status (77) because they simply can not work on Sailfish at the moment.

Thaodan commented:
Can you post those patches upstream that fix tests if bash isn't used?

rainemak commented:
> Can you post those patches upstream that fix tests if bash isn't used?

Not relevant for this task. @Tomin1 please create tasks to bz regarding upstreaming.

rainemak commented on rpm/0003-Move-large-array-out-of-stack.patch:
```diff
@@ -0,0 +1,31 @@
+From 353ba68a1c1184aeea1355abe0c156f053362620 Mon Sep 17 00:00:00 2001
```
This could be formatted as zero commit but fine like this as well.

Tomin1 replied:
Fixed.

spiiroin commented on rpm/0004-Fix-busybox-compatibility-for-test-execute.patch:
```diff
@@ -0,0 +1,43 @@
+ ExecStart=/usr/bin/test %C = /var/cache
+ ExecStart=/usr/bin/test %L = /var/log
+-ExecStart=/bin/sh -c 'test %u = $$(id -un 0)'
++ExecStart=/bin/sh -c 'test %u = root'
```
If the point is just to ensure this gets executed as root while avoiding literal "root": I guess this could be 'test %U = 0' too?

Tomin1 replied:
That %U = 0 would be the test just below.

Tomin1 replied:
Now it uses `getent` so it should be fine.

spiiroin replied:
:+1:

spiiroin commented on rpm/systemd-240-test-fix-tests-for-supplementary-groups.patch:
```diff
@@ -0,0 +1,152 @@
+ 
+ [Service]
+-ExecStart=/bin/sh -x -c 'test "$$(id -G)" = "1" && test "$$(id -g)" = "1" && test "$$(id -u)" = "1"'
++ExecStart=/bin/sh -x -c 'HAVE=0; for g in $$(id -G); do test "$$g" = "1" && HAVE=1; done; test "$$HAVE" -eq 1'
```
id -G | grep -E '\<1\>' comes to mind as slightly simpler way to do this, but as this is upstream patch... nevermind

spiiroin reviewed (approved):
Did not spot any obvious faults

spiiroin commented:
Hmm, that bug number... should it be 52786, not 52768?

Thaodan commented:
> Can you post those patches upstream that fix tests if bash isn't used?
> 
> I'm not sure they make sense for upstream. That `$$(id -un 0)` replacement
> with `root` doesn't consider that uid 0 could be named something else than
> root although in practice it almost never is so it depends a bit how they
> see it. (That said replacing id uses with getent + cut instead might be
> okay to them.) The other busybox "fix" is to skip some tests because I
> didn't come up with a good way to implement find -writable, so I doubt they
> would be interested in that.
> 

Ok I understand. 
Find -writeable could be mostly replaced with find -perm

Tomin1 commented:
> Find -writeable could be mostly replaced with find -perm

I wonder if it fits here. -writable checks for write access, -perm checks for permission bits but not the actual (computed) write access and the test wants to know if the files are writable AFAICT. But good idea anyway, just maybe not what I need here.

> Hmm, that bug number... should it be 52786, not 52768?

It's a mistake, I mixed the digits in my head. It should be 52786. I'll correct it. Thanks for mentioning it, as I surely wouldn't have spotted it myself.

Tomin1 commented:
Updated. Now commits have zero ids and I changed that root name test to use `getent` and `cut`. Also some comments were adjusted and since test-bfp works on Xperia 10 II, I removed skipped status from that as AFAIK I can't have two possible expected statuses (0 and 77). I added perl to tests package requires as that's needed for some tests.

The two remaining failing tests should be test-fd-util and test-cgroup, but they are likely to remain like that for now.

Tomin1 commented:
> Hmm, that bug number... should it be 52786, not 52768?

The id should be correct now.
